### ISSUE TITLE
Issues with "Dart" spell

### ISSUE THREAD

**Issue by alphec** [CatoThe1stElder/WFRP-4th-Edition-FoundryVTT#382]
There's an issue with dragging the Dart spell to the character sheet. An "Unrecognized identifier" exception is shown. With the dev tools, I was able to see that the exception is thrown in `actor-wfrp4e.js` when evaluating the formula `"+0+ willpower bonus"` in the `calculateSpellDamage(formula, isMagicMissile)` method.

In this method `+ willpower bonus` is added to the formula, but imho it should be resolved to a number.

IMHO this seems to be an issue with all magical missles.

**Comment by alphec** [comment]
Ok i found it. In the while loop, the if condition has to be adapted to use `WFRP4E` instead of the actor data.
```js
  calculateSpellDamage(formula, isMagicMissile)
  {
    let actorData = this.data
    formula = formula.toLowerCase();

    if (isMagicMissile) // If it's a magic missile, damage includes willpower bonus
    {
      formula += "+ willpower bonus"
    }

    // Iterate through characteristics
    for(let ch in actorData.data.characteristics)
    {
      // If formula includes characteristic name
      while (formula.includes(WFRP4E.characteristics[ch].toLowerCase()))
      {
        // Determine if it's looking for the bonus or the value
        if (formula.includes('bonus'))
          formula = formula.replace(WFRP4E.characteristics[ch].toLowerCase().concat(" bonus"),  actorData.data.characteristics[ch].bonus);
        else
          formula = formula.replace(WFRP4E.characteristics[ch].toLowerCase(),  actorData.data.characteristics[ch].value);
      }
    }

    return eval(formula);
  }
```

I fixed it locally for me and can prepare a PR if wanted. IMHO it needs further testing.

**Comment by moo-man** [comment]
Yeah because the actor template had these values hard coded previously, my update to the template to support translation doesn't affect old actors. A solution to this is to just duplicate the actor, but users don't know that, so using the config object would fix this issue, feel free to submit that PR.

**Comment by alphec** [comment]
I'm not sure about the duplication workaround. I have a brand new installation and added new chars when I encountered the bug. PR is coming up later today.

**Comment by alphec** [comment]
Hmm I think this is obsolete with https://github.com/CatoThe1stElder/WFRP-4th-Edition-FoundryVTT/commit/40724fee47f64422eef926dd07739de2e06ae9e8. Good!

**Comment by alphec** [comment]
Ah now I think what you meant with the "old actor"... I will provide a PR.

**Comment by moo-man** [comment]
Hi @alphec is this issue resolved?
